File: BOJ/10828(스택).py
#2022.01.14 10828(스택)문제


def push(x):                    #정수를 스택에 넣는 연산
    stack.append(x)

def pop():                      #스택 가장위에 정수를 빼고 그 수를 출력
    if len(stack) ==0:
        return-1

    if not stack:
        return -1
    else:
        return stack.pop()
    
def size():                     #스택에 들어있는 정수 개수 출력
    return len(stack)

def empty():                    #스택이 비어있으면1 아니면 0 출력
    if len(stack)==0:
        return 1
    else:
        return 0

def top():                      #가장 위의 정수를 출력, 없는경우 -1 출력
    return stack[-1] if stack else -1



import sys
N = int(input())   #이 개수만큼 반복
stack = []


#sys.stdin.readlin() => 반복문으로 여러줄 입력받을때는 input() 대신 사용
#                       여러줄에 input() 사용시 시간초과 문제
#rstrip()            => 오른쪽 공백 삭제 
#split()             => 문자열 분리
for _ in range(N):
    input_split = sys.stdin.readline().rstrip().split()
    cmd = input_split[0]
    if cmd == "push":
        push(input_split[1])
    elif cmd == "pop":
        print(pop())
    elif cmd == "size":
        print(size())
    elif cmd == "empty":
        print(empty())
    elif cmd == "top":
        print(top())



        

File: BOJ(JS)/10828(스택).js
const input = require("fs").readFileSync("10828.txt").toString().split("\n");

const iter = input.shift();

let stack = [];

// 시간 초과되는 것을 방지하기 위해
// ans 배열에 값들을 넣어놓고 마지막에 join으로 문자열로 바꿔서 한번에 출력
let ans = [];

// input 배열의 원소를 하나씩 짚으면서 명령어 판단
for (let i = 0; i < iter; i++) {
  if (input[i].includes("push")) {
    // "push 1", 이런 식으로 되어있기 때문에 split을 통해 정수를 분리해야함
    let pushNum = input[i].split(" ");
    stack.push(Number(pushNum[1]));
  } else if (input[i].includes("pop")) {
    // stack.length가 0보다 커야지 pop 가능
    // 스택 가장 위(stack 배열 가장 뒤)의 정수를 빼고, 출력해야함.
    if (stack.length > 0) {
      let popNum = stack.pop();
      ans.push(popNum);
    } else {
      ans.push(-1);
    }
  } else if (input[i].includes("size")) {
    // 단순 stack.length 출력
    ans.push(stack.length);
  } else if (input[i].includes("empty")) {
    // 비어있으면 1, 들어있으면 0 출력
    if (stack.length === 0) {
      ans.push(1);
    } else {
      ans.push(0);
    }
  } else if (input[i].includes("top")) {
    // 비어있으면 -1, 들어있으면 가장 뒤에 있는 값 출력
    if (stack.length === 0) {
      ans.push(-1);
    } else {
      ans.push(stack[stack.length - 1]);
    }
  }
}

console.log(ans.join("\n"));
